src/lib/components/worker-compatibility.svelte

```diff
@@ -0,0 +1,194 @@
+  {:else}
+    <tr class="w-full">
+      <td colspan="6">
+        <EmptyState title={translate('workers.no-version-sets-found')} />
+      </td>
+    </tr>
+  {/each}
+</Table>
+<h2 class="flex items-center gap-2 text-base font-medium" data-testid="workers">
+  {translate('workers.workers')}
+  <Badge type="count" class="rounded-sm">{workers?.pollers?.length || 0}</Badge>
+</h2>
+<Table class="mb-6 w-full min-w-[600px] table-fixed">
+  <caption class="sr-only" slot="caption"
+    >{translate('workflows.workers-tab')}</caption
+  >
+  <TableHeaderRow slot="headers">
+    <th class={reachability?.buildIdReachability?.length ? 'w-3/12' : 'w-6/12'}
+      >{translate('common.id')}</th
+    >
+    {#if reachability?.buildIdReachability?.length}
+      <th class="w-3/12">{translate('workers.version')}</th>
+      <th class="w-2/12">{translate('workers.retirability')}</th>
+    {/if}
+    <th class="w-2/12">{translate('workflows.last-accessed')}</th>
+    <th class="w-2/12">
+      <p class="text-center">
+        {translate('workflows.workflow-task-handler')}
+      </p>
+    </th>
+    <th class="w-2/12 text-center">
+      <p class="text-center">{translate('workflows.activity-handler')}</p>
+    </th>
+  </TableHeaderRow>
+  {#each workers?.pollers as poller (poller.identity)}
+    {@const buildId = getCurrentPollerBuildId(poller)}
+    {@const pollerReachability = getBuildIdReachability(
+      reachability,
+      taskQueue,
+      buildId,
+    )}
+    <TableRow data-testid="worker-row">
+      <td class="text-left" data-testid="worker-identity">
+        <p class="select-all">{poller.identity}</p>
+      </td>
+      {#if reachability?.buildIdReachability?.length}
+        <td class="text-left" data-testid="worker-identity">
+          <p class="select-all">
+            <CompatibilityBadge
+              defaultVersion={buildId === defaultVersion}
+              active={buildId === defaultVersion}
+              {buildId}
+            >
+              <svelte:fragment slot="overall-default-worker">
+                {#if buildId === defaultVersion}{translate(
+                    'workers.overall',
+                  )}{/if}
+              </svelte:fragment>
+              <svelte:fragment slot="default-worker">
+                {translate('workers.default')}
+              </svelte:fragment>
+            </CompatibilityBadge>
+          </p>
+        </td>
+        <td class="text-left" data-testid="worker-last-access-time">
+          <p>
+            <span class:reachability={!!pollerReachability}
+              >{pollerReachability}</span
+            >
+          </p>
+        </td>
+      {/if}
+      <td class="text-left" data-testid="worker-last-access-time">
+        <p class="select-all">
+          {formatDate(poller.lastAccessTime, $timeFormat, {
+            relative: $relativeTime,
+          })}
+        </p>
+      </td>
+      <td data-testid="workflow-poller">
+        {#if poller.taskQueueTypes.includes('WORKFLOW')}
```

Comment:
Since we use this repeatedly, it would be cool if it was its own component that took the boolean as a prop.

```
<PollerIcon state={poller.taskQueueTypes.includes('WORKFLOW')} />
```

And then internally, did the conditional to figure out whether or not to show a checkmark or close icon.

Reply:
👍